lib/fs-driver.js

```diff
@@ -173,3 +206,100 @@ FsDriver.prototype.sed = function (search, replace, name, line, cb) {
 FsDriver.prototype.exists = function (path) {
   return fs.existsSync(this.absolutePath(path));
 };
+
+/**
+ * Performs a file diff.
+ * @param {string} a Name of the first file for the diff.
+ * @param {string} b Name of the second file.
+ * @param {function} cb Callback to execute with the results of the diff.
+ */
+FsDriver.prototype.diff = function (a, b, cb) {
+  var command = [
+    'diff -u -r',
+    this.absolutePath(a),
+    this.absolutePath(b)
+  ].join(' ');
+  return this.exec(command, function (err, diff) {
+    // `diff` returns 1 when there are differences and > 1 when there is trouble
+    if (err && err.code > 1) { return cb(err); }
+    cb(null, diff);
+  });
+};
+
+/**
+ * Removes a file.
+ * @param {string} filename Name of the file to remove.
+ * @param {function} cb Callback to execute after the file remove.
+ */
+FsDriver.prototype.remove = function (filename, cb) {
+  return this.exec([
+    'rm',
+    this.absolutePath(filename)
+  ].join(' '), cb);
+};
+
+/**
+ * Recursively removes a file or directory.
+ * @param {string} filename Name of the file to remove.
+ * @param {function} cb Callback to execute after the file remove.
+ */
+FsDriver.prototype.removeRecursive = function (filename, cb) {
+  return this.exec([
+    'rm -rf',
+    this.absolutePath(filename)
+  ].join(' '), cb);
+};
+
+/**
+ * Attempts to find an available working directory relative to /tmp.
+ */
+FsDriver.prototype.findWorkingDirectory = function() {
+  var rootName = last(this.root.split('/'));
+  var workingPath = '/tmp/.' + rootName + '.fs-work';
+  var n = 0;
```

Comment:
+1 - while loops can be bad - while it's unlikely that many exist, using a random something generator may be a better move